src/components/DistanceEReceipt.tsx

```diff
@@ -1,53 +1,45 @@
-import lodashGet from 'lodash/get';
 import React, {useMemo} from 'react';
 import {ScrollView, View} from 'react-native';
-import _ from 'underscore';
 import EReceiptBackground from '@assets/images/eReceipt_background.svg';
 import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
-import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as CurrencyUtils from '@libs/CurrencyUtils';
 import * as ReceiptUtils from '@libs/ReceiptUtils';
 import * as ReportUtils from '@libs/ReportUtils';
 import * as TransactionUtils from '@libs/TransactionUtils';
 import tryResolveUrlFromApiRoot from '@libs/tryResolveUrlFromApiRoot';
+import {TranslationPaths} from '@src/languages/types';
+import {Transaction} from '@src/types/onyx';
+import {WaypointCollection} from '@src/types/onyx/Transaction';
 import Icon from './Icon';
 import * as Expensicons from './Icon/Expensicons';
 import ImageSVG from './ImageSVG';
 import PendingMapView from './MapView/PendingMapView';
 import Text from './Text';
 import ThumbnailImage from './ThumbnailImage';
-import transactionPropTypes from './transactionPropTypes';
 
-const propTypes = {
+type DistanceEReceiptProps = {
     /** The transaction for the distance request */
-    transaction: transactionPropTypes,
+    transaction: Transaction;
 };
 
-const defaultProps = {
-    transaction: {},
-};
-
-function DistanceEReceipt({transaction}) {
-    const theme = useTheme();
+function DistanceEReceipt({transaction}: DistanceEReceiptProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
-    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction) : {};
-    const {amount: transactionAmount, currency: transactionCurrency, merchant: transactionMerchant, created: transactionDate} = ReportUtils.getTransactionDetails(transaction);
+    const {thumbnail = ''} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction) : {};
+    const {amount: transactionAmount, currency: transactionCurrency, merchant: transactionMerchant, created: transactionDate} = ReportUtils.getTransactionDetails(transaction) ?? {};
     const formattedTransactionAmount = transactionAmount ? CurrencyUtils.convertToDisplayString(transactionAmount, transactionCurrency) : translate('common.tbd');
-    const thumbnailSource = tryResolveUrlFromApiRoot(thumbnail || '');
-    const waypoints = lodashGet(transaction, 'comment.waypoints', {});
-    const sortedWaypoints = useMemo(
+    const thumbnailSource = tryResolveUrlFromApiRoot((thumbnail as string) || '');
```

Comment:
I think you may need to adjust `tryResolveUrlFromApiRoot` function, as `thumbnail` can be also `ImageSourcePropType` type and it's not handled properly.

Reply:
I know @blazejkustra that's why I commented on the pr **Details** section. As it could be a bug unrelated with the ts migration (`Trying to pass a variable which can be ImageSourcePropType to tryResolveUrlFromApiRoot which expects a string `). If you can confirm it's not a problem we can update  `tryResolveUrlFromApiRoot` accordingly 👍

Reply:
Go ahead and adjust `tryResolveUrlFromApiRoot` as part of Ts migration, should be fairly easy? 🤔

Reply:
So to confirm, should I update `tryResolveUrlFromApiRoot` to accept `ImageSourcePropType` too as an argument?

Reply:
Yes, and handle it accordingly

Reply:
`getThumbnailAndImageURIs` is typed to also return thumbnail of type `ImageSourcePropType`
https://github.com/Expensify/App/blob/a799adfd577da0a8117cebdb402e675de2bd3625/src/libs/ReceiptUtils.ts#L14-L15
It is supposed to return non-string thumbnail when transaction pending waypoints
https://github.com/Expensify/App/blob/a799adfd577da0a8117cebdb402e675de2bd3625/src/libs/ReceiptUtils.ts#L40-L41
But I checked that even in that case it returns a url string instead of `ImageSourcePropType`.
So we have two options here:
1. keep the current change `    const thumbnailSource = tryResolveUrlFromApiRoot((thumbnail as string) || '');`
2. Change `getThumbnailAndImageURIs` to have return type of only string
What do u think?? @shubham1206agra @blazejkustra @VickyStash

Reply:
> Change getThumbnailAndImageURIs to have return type of only string
What do u think??

How would you do it, currently it returns thumbnail as ImageSourcePropType:
```ts
let image = ReceiptGeneric;
    if (fileExtension === CONST.IOU.FILE_TYPES.HTML) {
        image = ReceiptHTML;
    }

    if (fileExtension === CONST.IOU.FILE_TYPES.DOC || fileExtension === CONST.IOU.FILE_TYPES.DOCX) {
        image = ReceiptDoc;
    }

    if (fileExtension === CONST.IOU.FILE_TYPES.SVG) {
        image = ReceiptSVG;
    }

    const isLocalFile = typeof path === 'number' || path.startsWith('blob:') || path.startsWith('file:') || path.startsWith('/');
    return {thumbnail: image, image: path, isLocalFile};
```

Reply:
>How would you do it, currently it returns thumbnail as ImageSourcePropType:

Nope. It returns a string url something like this `/1b96fa279dcd8a80ee9f.png` for thumbnail. I think it is a mistake we made before assuming the thumbnails are `ImageSourcePropType`.
For instance, if you create a request with pdf receipt and check what is being returned here you can proof it
https://github.com/Expensify/App/blob/55746c78a6ef59d9ad636eda8c610023c3cd62ac/src/components/ReportActionItem/MoneyRequestPreview.js#L178-L179
please @grgia @shubham1206agra let's deal with it and resolve it. It is holding the pr forever. 🙏

Reply:
Did you check that it is a string on all platforms?

Reply:
I tested it, and `ImageSourcePropType` looks right to me, on ios/android such assets are not strings. 

I don't understand one thing, why not adjust `tryResolveUrlFromApiRoot`, so it handles string, number and ImageSourcePropType gracefully?

```ts
export type ImageSourcePropType =
  | ImageURISource
  | ImageURISource[]
  | ImageRequireSource;
```

Reply:
For native thumbnail it is `number` which is the case we are handling here 
https://github.com/Expensify/App/blob/55746c78a6ef59d9ad636eda8c610023c3cd62ac/src/libs/tryResolveUrlFromApiRoot.ts#L25-L27
Had it been `ImageSourcePropType` in fact it would have caused an error. So we should fix thumbnail type to be string or number.
https://github.com/Expensify/App/blob/55746c78a6ef59d9ad636eda8c610023c3cd62ac/src/libs/ReceiptUtils.ts#L14-L15

Reply:
> Had it been ImageSourcePropType in fact it would have caused an error. 

@FitseTLT `number` is part of `ImageRequireSource`, so the type is correct and it doesn't cause an error 😅 
```ts
export type ImageSourcePropType =
  | ImageURISource
  | ImageURISource[]
  | ImageRequireSource;

export type ImageRequireSource = number;
```

@VickyStash @fabioh8010 @kubabutkiewicz what's your opinion on this?

Reply:
> How would you do it, currently it returns thumbnail as ImageSourcePropType, if the type is wrong, let's fix it 😄

@FitseTLT How about changing global.d.ts to something like this:
```ts
declare module '*.png' {
    const value: string | number;
    export default value;
}
// same for .jpg?
```

Reply:
Or fix `getThumbnailAndImageURIs` like you suggested above:
> Change getThumbnailAndImageURIs to have return type of only string
What do u think?

Reply:
Lets not change `getThumbnailAndImageURIs`. I will suggest to separate out the `getThumbnailAndImageURIs` logic into new function for EReceipts maybe. Or fix the component to accept `ImageSourcePropType` correctly.

Reply:
bump @FitseTLT

Reply:
@blazejkustra
> I will suggest to separate out the getThumbnailAndImageURIs logic into new function for EReceipts maybe

We can use overload signature as there is no different logic to handle inside getThumbnailAndImageURIs

>Or fix the component to accept ImageSourcePropType correctly

We have checked that getThumbnailAndImageURIs returns thumnail as `string|number` and it is working fine so far with tryResolveUrlFromApiRoot I don't understand why we should be tweaking the logic on a Ts migration issue.

Reply:
> I don't understand why we should be tweaking the logic on a Ts migration issue.

Because types are wrong 🤷‍♂️ If we don't fix it, we'll run int this issue in other components as well.

> We can use overload signature as there is no different logic to handle inside getThumbnailAndImageURIs

Give it a try!

Reply:
I saw all instances we used the `thumbnail` and `image` returned from `getThumbnailAndImageURIs` and we are passing it to `source` of `Image` component as an object where the `uri` prop is set to `thumbnail` or `image` returned from `getThumbnailAndImageURIs` so thumbnail can not be `| ImageURISource | ImageURISource[]`
https://github.com/Expensify/App/blob/a799adfd577da0a8117cebdb402e675de2bd3625/src/components/ReportActionItem/ReportActionItemImage.js#L75-L76
So what I recommend is to change 
https://github.com/Expensify/App/blob/a799adfd577da0a8117cebdb402e675de2bd3625/src/libs/ReceiptUtils.ts#L12-L14
```

    image: ImageRequireSource | string;
    thumbnail: ImageRequireSource | string | null;
```

Reply:
@blazejkustra @shubham1206agra let's not worry about this as I am working on an [issue](https://github.com/Expensify/App/issues/28827) that will migrate to EReceiptThumnails and `getThumbnailAndImageURIs` is not going to return thumnails anymore 👍

Reply:
ping @blazejkustra @shubham1206agra on ^